pkgs/development/libraries/glibc/default.nix

```diff
@@ -75,11 +75,14 @@ callPackage ./common.nix { inherit stdenv; } {
     # - clang-wrapper in cross-compilation
     # Last attempt: https://github.com/NixOS/nixpkgs/pull/36948
     preInstall = ''
-      if [ -f ${stdenv.cc.cc}/lib/libgcc_s.so.1 ]; then
+      if [ -f ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so.1 ]; then
           mkdir -p $out/lib
-          cp ${stdenv.cc.cc}/lib/libgcc_s.so.1 $out/lib/libgcc_s.so.1
+          cp ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so.1 $out/lib/libgcc_s.so.1
           # the .so It used to be a symlink, but now it is a script
-          cp -a ${stdenv.cc.cc}/lib/libgcc_s.so $out/lib/libgcc_s.so
+          cp -a ${lib.getLib stdenv.cc.cc}/lib/libgcc_s.so $out/lib/libgcc_s.so
+          # wipe out reference to previous libc it was built against
+          chmod +w $out/lib/libgcc_s.so.1
+          patchelf --set-rpath $out/lib $out/lib/libgcc_s.so.1
```

Comment:
```suggestion
```

I don't think we should do that.  It bypasses this important alarm system:

https://github.com/NixOS/nixpkgs/blob/31819ec0e7ba7c0e382fe5cc2a64af954dcaa6a3/pkgs/stdenv/linux/default.nix#L511

Reply:
> Do we really need to nuke the references?

I'm pretty sure we do not need to.

I rebased on this PR but reverted the three lines above in 6ddfd418b2d9dc8fa4e6954274e15e9890218ea1.  The [result](https://github.com/NixOS/nixpkgs/pull/209870) builds correctly at 9c7e9ef256714455914180c0bcb434e014e5b75a on `x86_64` and `powerpc64le`.

Reply:
`github` say that commit is not a part of any branch. Which `gcc` your branch copies `libgcc_s.so.1`? `bootstrapTools` one of built against available `glibc`?

`bootstrapTools` one already has references nuked as part of final tarball generation.

If we use local `gcc` that does contain reference to `glibc` it was built with (like one added in https://github.com/NixOS/nixpkgs/pull/209063). If it's copied from first `gcc` built by `bootstrapTools` `glibc` then reference checker detects it as:

```
$ nix build --no-link -f. stdenv
error: output '/nix/store/6f8pxzlxkd67ph679axs2ghsxdgsca03-stdenv-linux' is not allowed to refer to the following paths:
         /nix/store/g23g6pxvfilnx50ray41ry35d5vmx5dj-bootstrap-tools
         /nix/store/lan2w3ab1mvpxj3ppiw2sizh8i7rpz7s-busybox
         /nix/store/nm7fxi26avig3arwpasfy0avfwyb41i3-bootstrap-stage0-glibc-bootstrap
```

I think fixes like https://github.com/NixOS/nixpkgs/pull/209153 might affect library lookup order by using correct gcc's library and make it more prominent.

Reply:
This is resolved; I was wrong about this.